spec/supabase_js_v2.yml

```
@@ -421,6 +421,80 @@ pages:
             if (event == 'PASSWORD_RECOVERY') console.log('PASSWORD_RECOVERY', session)
           })
           ```
+  auth.mfa.enroll():
+    title: 'mfa.enroll()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.enroll'
+    notes: |
+      - Currently, `totp` is the only supported `factorType`.
+      - To generate a QR code containing the `totp` secret, 
+      - To verify the factor next, see `auth.mfa.challengeAndVerify()`.
+      - To generate a QR code for the `totp` secret in nextjs, you can do the following:
+      ```html
+      <Image src={data.totp.qr_code} alt={data.totp.uri} layout="fill"></Image>
+      ```
+    examples:
+      - name: Enroll a TOTP factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.enroll({ 
+            factorType: 'totp' 
+          })
+          ```
+  auth.mfa.challenge():
+    title: 'mfa.challenge()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.challenge'
+    examples:
+      - name: Creates a challenge for a factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.challenge({ 
+            factorId: '00000000-0000-0000-0000-000000000000' 
+          })
+          ```
+  auth.mfa.verify():
+    title: 'mfa.verify()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.verify'
+    examples:
+      - name: Verifies a challenge for a factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.verify({ 
+            factorId: '00000000-0000-0000-0000-000000000000', 
+            challengeId: '11111111-1111-1111-1111-111111111111', 
+            code: '123456'
+          })
+          ```
+  auth.mfa.challengeAndVerify():
+    title: 'mfa.challengeAndVerify()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.challengeAndVerify'
+    notes: |
+      - Combines `mfa.challenge()` and `mfa.verify()` in 1 step
+    examples:
+      - name: Creates and verifies a challenge for a factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.challengeAndVerify({
+            factorId: '00000000-0000-0000-0000-000000000000',
```

/nit same thing about uuid
```suggestion
            factorId: '34e770dd-9ff9-416c-87fa-43b31d7ef225',
```